pkg/unidling/controller/unidling_controller.go

```diff
@@ -98,17 +96,10 @@ func NewUnidlingController(scaleNS scale.ScalesGetter, mapper meta.RESTMapper, e
 	}
 
 	_, controller := cache.NewInformer(
-		&cache.ListWatch{
-			// No need to list -- we only care about new events
-			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
-				options.FieldSelector = fieldSelector.String()
-				return evtNS.Events(metav1.NamespaceAll).List(context.TODO(), options)
-			},
-			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
-				options.FieldSelector = fieldSelector.String()
-				return evtNS.Events(metav1.NamespaceAll).Watch(context.TODO(), options)
-			},
-		},
+		// Bug 1887745: avoiding wiring up cache.ListWatcher instance, and avoiding listing all
+		// events from all namespaces, and create a race condition on big clusters, where the amount
+		// of resources may not be enough to handle all events at once.
+		nil,
```

Comment:
were you able to confirm that you still got add events with this @otaviof ? 

and is it that with this change, do you no longer got the relists on the the `resyncPeriod` interval ?

Reply:
@gabemontero yes, positve. It does handle new events. And also, it does not issue `LIST` on `resyncPeriod` and during client bootstrap.

Reply:
OK good.  Then I suggest updating that comment to note a) we only care about add events, and b) this form of informer initialization will still get us add events, but avoid lists on the resync interval etc.

Reply:
Sure, thanks for the suggestion. Please consider updated description.

Reply:
I would put the full url to the bugzilla, "Bug <number>" seems to ambiguous.

Reply:
@coreydaley okay, I'm adding the full Bugzilla URL in the comment. Please consider.

Reply:
can you add the details I asked for in https://github.com/openshift/openshift-controller-manager/pull/145/files#r516818486 @otaviof 

`a) we only care about add events, and b) this form of informer initialization will still get us add events, but avoid lists on the resync interval etc.`

Reply:
@gabemontero It was added to the PR description, but now also as a comment as well. Duly noted.

Reply:
/lgtm

/assign @adambkaplan 
(though he may be out for a bit ... but I'll assign him in case he sees this and the discussion above prompts him to approve)
/assign @deads2k 
(as `git blame` cites him *from 5 years ago* for the prior line 102 `// No need to list -- we only care about new events` ... ideally we get a sanity check from him that our interpretation of things is correct)

Reply:
Thank you, folks!

Reply:
How does this watch without a ListWatch?

Reply:
@sttts in fact it would not properly watch without a `WatchFunc` in the `cache.ListWatch`. The attempt before was removing too many elements from this Informer.